### PR TITLE
Add assume role option to boto3 driver

### PR DESCRIPTION
**Pull request by sfc-gh-afedorov**
tested against our local and dev account [here](https://app.snowflake.com/us-west-2/snowhouse/w3RZuCaZvmW3)

### CODE REVIEW

File: lambda_src/drivers/process_boto3.py [hunk 1]
```diff
@@ -3,21 +3,53 @@
...
+from utils import pick
 
 DISALLOWED_CLIENTS = {'kms', 'secretsmanager'}
 
 
 def process_row(
-    client_name, method_name, results_path=None, region='us-west-2', **kwargs
+    client_name,
+    method_name,
+    assume_role_arns=None,
```
Comment (sfc-gh-pkommini) [inline lambda_src/drivers/process_boto3.py:14]: Is this always ARNs? Is it always a list? What possible values can this variable take?

If it is a sequence of roles that need to be assumed would it make sense to add the keyword `sequence` or `chain` in the variable to indicate that?

**Review by sfc-gh-pkommini — APPROVED** [review]
lgtm but for cosmetic variable name clarity.

File: lambda_src/drivers/process_boto3.py [hunk 2]
```diff
@@ -3,21 +3,53 @@
 import boto3
 from botocore.response import StreamingBody
 
+from utils import pick
```
Comment (sfc-gh-pkommini) [inline lambda_src/drivers/process_boto3.py:6]: This should be a relative import.

Comment (sfc-gh-afedorov) [reply lambda_src/drivers/process_boto3.py]: good catch -- was testing without top level package

File: lambda_src/drivers/process_boto3.py (hunk 1 above)
Comment (sfc-gh-afedorov) [reply lambda_src/drivers/process_boto3.py:14]: it's a sequence of roles, yes, with either strings or objects which are going to be passed into `assume_role`, e.g. `{"RoleArn":"...","ExternalId":"..."}`

**Comment by sfc-gh-afedorov** [comment]
tested also using `make build`

**Pull request merged**